syntax = "proto2";
package QtProto;

message QVector3D {
    required float x = 1;
    required float y = 2;
    required float z = 3;
}

message QVector2D {
    required float x = 1;
    required float y = 2;
}

message QPointF {
    required double x = 1;
    required double y = 2;
}

//For backwards compatiblity
message QString {
    required bytes stringData = 1;
}

message QDate {
    required int32 year = 1;
    required int32 month = 2;
    required int32 day = 3;
}

message QSize {
    required int32 width = 1;
    required int32 height = 2;
}

message QSizeF {
    required double width = 1;
    required double height = 2;
}

message QStringList {
    repeated QString legacy_strings = 1;
}

message QUuid {
    required bytes uuid = 1;  // store QUuid as 16 raw bytes
}
